src/Plugin/RulesAction/EntityCreateDeriver.php

```diff
@@ -89,9 +89,19 @@ public function getDerivativeDefinitions($base_plugin_definition) {
           continue;
         }
 
+        // If this is an entity reference then we expect the target type as
+        // context.
+        if ($definition->getType() === 'entity_reference') {
+          $type = 'entity:' . $definition->getSetting('target_type');
```

Comment:
This hard-codes two things it shouldn't / hasn't:
- entity_references do not have to be of type 'entity_reference'. Let's check for a data reference interface instead.
- This should not hard-code the setting-name, which is internal to a specific field type. It should use the getTargetDefintion (or how it's called) and call getDataType() on it.